rasa/importers/importer.py

```diff
@@ -0,0 +1,247 @@
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingFileImporter:
+    """Common interface for different mechanisms to load training data."""
+
+    async def get_domain(self) -> Domain:
+        """Retrieves the domain which should be used for the training."""
+        raise NotImplementedError()
+
+    async def get_story_data(
+        self,
+        interpreter: "NaturalLanguageInterpreter" = RegexInterpreter(),
+        template_variables: Optional[Dict] = None,
+        use_e2e: bool = False,
+        exclusion_percentage: Optional[int] = None,
+    ) -> StoryGraph:
+        """Retrieves the story data which should be used for the training."""
+
+        raise NotImplementedError()
+
+    async def get_config(self) -> Dict:
+        """Retrieves the configuration which should be used for the training."""
+        raise NotImplementedError()
+
+    async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
+        """Retrieves the nlu training data which should be used for the training."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def load_from_config(
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+        load_only_nlu_data: bool = False,
+    ) -> "TrainingFileImporter":
+        """Loads a `TrainingFileImporter` instance from a configuration file."""
+
+        config = io_utils.read_config_file(config_path)
+        return TrainingFileImporter.load_from_dict(
+            config, config_path, domain_path, training_data_paths, load_only_nlu_data
+        )
+
+    @staticmethod
+    def load_from_dict(
+        config: Optional[Dict],
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+        load_only_nlu_data: bool = False,
+    ) -> "TrainingFileImporter":
+        """Loads a `TrainingFileImporter` instance from a dictionary."""
+        config = config or {}
+        importers = config.get("importers", [])
+        importers = [
+            TrainingFileImporter._importer_from_dict(
+                importer, config_path, domain_path, training_data_paths
+            )
+            for importer in importers
+        ]
+        importers = [importer for importer in importers if importer]
+
+        if not importers:
+            importers = [
+                SimpleFileImporter(config_path, domain_path, training_data_paths)
+            ]
+
+        importer = CombinedFileImporter(importers)
+        if load_only_nlu_data:
+            importer = NluFileImporter(importer)
+
+        return importer
+
+    @staticmethod
+    def _importer_from_dict(
+        importer_config: Dict,
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+    ) -> Optional["TrainingFileImporter"]:
+        from rasa.importers.skill import SkillSelector
```

Comment:
Why is the `SkillSelector` in a different file if it is also an importer?

Reply:
I wanted to avoid to having one large module. Would you suggest a different split?

Reply:
Yeah, make sense. But maybe we should then move also the `SimpleFileImporter` to a separate file as it the only "real" importer next to the `SkillSelector`.

Reply:
👍